File: python/KidsWiththeGreatestNumberofCandies.py
# 1431. Kids With the Greatest Number of Candies

# Example 1:

# Input: candies = [2,3,5,1,3], extraCandies = 3
# Output: [true,true,true,false,true]
# Explanation: If you give all extraCandies to:
# - Kid 1, they will have 2 + 3 = 5 candies, which is the greatest among the kids.
# - Kid 2, they will have 3 + 3 = 6 candies, which is the greatest among the kids.
# - Kid 3, they will have 5 + 3 = 8 candies, which is the greatest among the kids.
# - Kid 4, they will have 1 + 3 = 4 candies, which is not the greatest among the kids.
# - Kid 5, they will have 3 + 3 = 6 candies, which is the greatest among the kids.


class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        highest = max(candies)
        values = []
        for i in range(0, len(candies)):
            reference = candies[i] + extraCandies
            if reference >= highest:
                values += [True]
            else:
                values += [False]
        return values

File: java/Kids With the Greatest Number of Candies.java
/*
 Kids With the Greatest Number of Candies - 1431

 There are n kids with candies. You are given an integer array candies, where each candies[i] 
 represents the number of candies the ith kid has, and an integer extraCandies, denoting the number 
 of extra candies that you have.

Return a boolean array result of length n, where result[i] is true if, 
after giving the ith kid all the extraCandies, they will have the greatest number of candies 
among all the kids, or false otherwise.

Note that multiple kids can have the greatest number of candies.
 */

import java.util.List;
import java.util.ArrayList;

 class Solution {

    // Function to determine which kids can have the greatest number of candies
    // after they receive an additional amount of extraCandies.
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {

        // Find the maximum number of candies that any kid currently has.
        int maxCandies = Integer.MIN_VALUE;
        
        for (int candy : candies) {
            maxCandies = Math.max(maxCandies, candy);
        }

        // List to store the results, whether each kid can have the maximum number
        // of candies after receiving extraCandies.
        List<Boolean> result = new ArrayList<>();

        // Loop through each kid's candies to determine if they can reach maxCandies
        // with their current candies plus extraCandies.
        for (int candy : candies) {
            // If the current kid's candies plus extraCandies is greater than or
            // equal to maxCandies, add 'true' to the result list, otherwise add 'false'.
            result.add(candy + extraCandies >= maxCandies);
        }

        // Return the result list.
        return result;
    }
}
